### ISSUE TITLE
Simplified New-PSFSupportPackage or a switch

### ISSUE THREAD

**Issue by Splaxi** [PowershellFrameworkCollective/psframework#165]
Hi,

Just starting getting my hands dirty with the module and it is great stuff. I would suggest making either a simplified version of New-PSFSupportPackage or provide a switch to limit what is included into a support package.

For my use case I would mostly only be interested in getting all the messages, and not more than that.

Brain storming ideas:
New-PSFSupportPackage -MessagesOnly
New-PSFSupportPackage -Include Messages,ConsoleDetails
New-PSFSupportPackage -Exclude SystemDetails

Any flavor that makes it possible to limit, include or exclude some of the collecting would be great. I'll be happy to help making the changes to the cmdlet/function, if we can agree on how to implement them while respecting the "no-breaking-changes" mindset. Having a -All option (or similar) should help with the "no-breaking-changes".

**Comment by FriedrichWeinmann** [comment]
Heya,

thanks for your kind words :)
The thing about "No breaking changes" pretty much means that `-All` is out the door, since that is the current default and thus means will be what must happen when a user runs the command without further selection.

That said, I like the idea about `-Include` and `-Exclude`. I'd implement that list of pieces of information to be packaged by using a flags enumeration  in C#. That would make it a lot easier to combine multiple settings into a cover string (Like how file system permissions can be combined into "Modify" when it's actually a set of permissions, `Default` could represent the default options while the user can pick them individually or combine them.

**Comment by FriedrichWeinmann** [comment]
> First off, the -All, as no breaking changes could be solved like this:
[switch] $All = [switch]::Present

But what is the benefit here? What happens when the user runs `-All:$false`? No Output?
No, your `-Include` and `-Exclude` parameters sound like exactly the way to go.

You can totally do the full implementation on this by ...
- First implementing the flags enumeration in C# (it's not as scary as it might sound, see example below)
- Then add the parameters and for each piece of information gathered, check whether it should be gathered first by checking the `$Include` and `$Exclude` parameters.

Now, for a flags enum, let's take a look at a piece of example code from the PSFramework:
```c#
using System;
[Flags]
public enum ConfigScope
{
    UserDefault = 1,
    UserMandatory = 2,
    SystemDefault = 4,
    SystemMandatory = 8,
}
```
As you can see, we assign numbers to pieces of text here. In PowerShell, think of this as an elaborate ValidateSet. Only it's better, because it allows us to create aliases or combine values like this:
```c#
using System;
[Flags]
public enum ConfigScope
{
    UserDefault = 1,
    UserMandatory = 2,
    SystemDefault = 4,
    BothDefault = 5,
    SystemMandatory = 8,
}
```
In a condition, you can now test whether the user picked `SystemDefault` - either directly or by choosing `BothDefault` - by doing binary comparison (Which is a bit ugly for the uninitiated but produces amazingly clean code):
```powershell
if ([ConfigScope]::BothDefault -band 4) { "Was Set" }
if ([ConfigScope]::SystemDefault -band 4) { "Was Set, too" }
if ([ConfigScope]::UserDefault -band 4) { "Was not set, so this text should not show" }
```

It's too late for me tonight, but if you want to take a look at this together tomorrow, I'll be happy to do so, those are amazingly useful in PowerShell and actually fairly fast to learn and apply.

**Comment by FriedrichWeinmann** [comment]
Implemented in 0.9.25.107
